samples/backend/python/quart/__init__.py

```diff
@@ -0,0 +1,181 @@
+
+def reconstruct_multipart_request(form: dict, files: dict):
+    json_content = form["json"]
+    json_dict = json.loads(json_content)
+    chat_request = AIChatRequest.from_dict(json_dict)
+
+    file_positions = sorted([get_file_position(file_key) for file_key in files])
+    for message_index, file_index, file_key in file_positions:
+        file = files[file_key]
+
+        if len(chat_request.messages) <= message_index:
+            raise ValueError(f"Invalid message index: {file_key}")
+
+        if chat_request.messages[message_index].files is None:
+            chat_request.messages[message_index].files = []
+
+        if len(chat_request.messages[message_index].files) != file_index:
+            raise ValueError(f"Invalid file index: {file_key}")
+
+        chat_request.messages[message_index].files.append(
+            AIChatFile(content_type=file.content_type, data=file.read())
+        )
+    return chat_request
+
+
+def to_openai_message(chat_message: AIChatMessage):
+    if chat_message.files is None:
+        return {
+            "role": chat_message.role.value,
+            "content": chat_message.content,
+        }
+
+    def encode_file_to_data_url(file: AIChatFile):
+        base64_data = base64.b64encode(file.data).decode("utf-8")
+        return f"data:{file.content_type};base64,{base64_data}"
+
+    images = [
+        {"type": "image_url", "image_url": {"url": encode_file_to_data_url(file)}}
+        for file in chat_message.files
+        if file.content_type.startswith("image/")
+    ]
+    return {
+        "role": chat_message.role.value,
+        "content": [{"type": "text", "text": chat_message.content}] + images,
+    }
+
+
+@app.route("/api/chat/", methods=["POST"])
+async def process_message():
+    try:
+        if request.content_type.startswith("multipart/form-data"):
+            form = await request.form
+            files = await request.files
+            chat_request = reconstruct_multipart_request(form, files)
+        elif request.content_type.startswith("application/json"):
+            chat_request_dict = await request.get_json()
+            chat_request = AIChatRequest.from_dict(chat_request_dict)
+        else:
+            return jsonify({"error": "Unsupported Media Type"}), 415
+        completion = await client.chat.completions.create(
+            model=AZURE_OPENAI_DEPLOYMENT,
+            messages=[to_openai_message(message) for message in chat_request.messages],
+        )
+
+        message = completion.choices[0].message
+        response = AIChatCompletion(
+            message=AIChatMessage(
+                role=AIChatRole(message.role),
+                content=message.content,
+            )
+        )
+        return jsonify(response.to_dict())
+    except Exception as e:
+        return (
+            jsonify(AIChatError(code="internal_error", message=str(e)).to_dict()),
+            500,
+        )
+
+
+def object_to_json_line(obj):
+    return f"{json.dumps(obj.to_dict())}\r\n"
```

Comment:
I recommend adding ensure_ascii=False to json.dumps() call
[per suggestion of the json-lines package maintainer that I talked with for my samples]

Reply:
will do

Reply:
now that we moved to use pydantic, this does not seem like something we need.